rust_src/src/buffers.rs

```diff
@@ -1843,4 +1847,80 @@ pub fn byte_char_debug_check(b: LispBufferRef, charpos: isize, bytepos: isize) {
     }
 }
 
+// True if B can be used as 'other-than-BUFFER' buffer.
+fn candidate_buffer(b: LispObject, buffer: LispObject) -> bool {
+    if let Some(buf) = Option::<LispBufferRef>::from(b) {
+        !b.eq(buffer) && buf.is_live() && !buf.is_hidden()
+    } else {
+        false
+    }
+}
+
+/// Return most recently selected buffer other than BUFFER.
+/// Buffers not visible in windows are preferred to visible buffers, unless
+/// optional second argument VISIBLE-OK is non-nil.  Ignore the argument
+/// BUFFER unless it denotes a live buffer.  If the optional third argument
+/// FRAME specifies a live frame, then use that frame's buffer list instead
+/// of the selected frame's buffer list.
+///
+/// The buffer is found by scanning the selected or specified frame's buffer
+/// list first, followed by the list of all buffers.  If no other buffer
+/// exists, return the buffer `*scratch*' (creating it if necessary).
+#[lisp_fn(min = "0")]
+pub fn other_buffer(buffer: LispObject, visible_ok: LispObject, frame: LispObject) -> LispObject {
+    let f: LispFrameRef = LispFrameLiveOrSelected::from(frame).into();
+    let pred = f.buffer_predicate;
+    let mut notsogood = None;
+
+    let frame_bufs = f
+        .buffer_list
+        .iter_cars(LispConsEndChecks::off, LispConsCircularChecks::off);
+    let all_bufs = LiveBufferIter::new().map(|x| x.into());
```

Comment:
This `into` might be better written using a `From` so the intent is clearer.

Reply:
Done.